components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -8482,6 +8483,36 @@ public APIProductIdentifier getAPIProductIdentifierFromUUID(String uuid) throws
         return identifier;
     }
 
+    /**
+     * Get API UUID by the API Identifier.
+     *
+     * @param identifier     API Identifier
+     * @param organizationId Identifier of an Organization
+     * @return String UUID
+     * @throws APIManagementException if an error occurs
+     */
+    public String getUUIDFromIdentifierMatchingOrganization(APIIdentifier identifier, String organizationId)
+            throws APIManagementException {
+        String uuid = null;
+        String sql = SQLConstants.GET_UUID_BY_IDENTIFIER_SQL_MATCHING_ORGANIZATION;
+        try(Connection connection = APIMgtDBUtil.getConnection()) {
+            PreparedStatement prepStmt = connection.prepareStatement(sql);
```

Comment:
shall we get prepStmt also into the try with resources block like `connection`? Otherwise this will not be closed properly

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/10036/commits/1fd1ec5697910cef0ee451b0a8d79e9d01623f2b